doc/rtd/reference/custom_modules/custom_configuration_module.rst

```diff
@@ -0,0 +1,18 @@
+.. _custom_configuration_module:
+
+Custom Configuration Module
+***************************
+
+Custom 3rd-party out-of-tree configuration modules can be added to cloud-init
+by:
+
+#. :ref:`Implement a config module<module_creation>` in a Python file with its
```

Comment:
I'm wondering if the referenced page wouldn't make more sense here than in the dev docs. What do you think? There may be some overlap in the intent, but since the intent is that users should be able to do this I think that having it under the "custom_modules" umbrella makes more sense. I think that as far as contributing goes, there are probably more useful things for new contributors to be thinking about than writing custom configuration modules for niche purposes.

Reply:
Yes, I think it is okay either way because:

That content is in the gray are where the persona is an advanced user that is consuming cloud-init as a user but is giving input to cloud-init that requires develoment knowledge about cloud-init's internal.

The same question applies for the documentation on how to create a new datasource and for https://github.com/canonical/cloud-init/issues/5559.